src/StatusQ/Controls/StatusPasswordInput.qml

```diff
@@ -0,0 +1,145 @@
+import QtQuick 2.14
+import QtQuick.Layouts 1.12
+import QtQuick.Controls 2.14
+
+import StatusQ.Core 0.1
+import StatusQ.Core.Theme 0.1
+import StatusQ.Controls 0.1
+
+/*!
+   \qmltype StatusPasswordInput
+   \inherits Item
+   \inqmlmodule StatusQ.Controls
+   \since StatusQ.Controls 0.1
+   \brief The StatusPasswordInput control provides a generic user password input with an option to display signing phrase
+
+   Example of how to use it:
+
+   \qml
+        StatusPasswordInput {
+            signingPhrase: "orange hello cygnet"
+            placeholderText: qsTr("Password")
+        }
+   \endqml
+
+   For a list of available components see StatusQ.
+*/
+
+Item {
+    id: root
+
+    /*!
+        \qmlproperty string StatusPasswordInput::signingPhrase
+        This property sets the signingPhrase of TextField field and signing seed phrase.
+    */
+    property string signingPhrase: ""
+
+    /*!
+        \qmlproperty int StatusPasswordInput::minimumHeight
+        This property sets the pixelSize of TextField field and signing seed phrase.
+    */
+    property int pixelSize: 15
+
+    /*!
+        \qmlproperty color StatusPasswordInput::selectionColor
+        This property sets the selectionColor property of TextField.
+    */
+    property color selectionColor: Theme.palette.primaryColor2
+
+    /*!
+        \qmlproperty color StatusPasswordInput::selectedTextColor
+        This property sets the selectedTextColor property of TextField.
+    */
+    property color selectedTextColor: Theme.palette.directColor1
+
+    /*!
+        \qmlproperty color StatusPasswordInput::color
+        This property sets the color property of TextField.
+    */
+    property color color: Theme.palette.directColor1
+
+    /*!
+        \qmlproperty alias StatusPasswordInput::placeholderText
+        This property holds a reference to the TextField's placeholderText property.
+    */
+    property alias placeholderText: input.placeholderText
+
+    /*!
+        \qmlproperty color StatusPasswordInput::placeholderTextColor
+        This property sets the placeholderTextColor property of TextField.
+    */
+    property color placeholderTextColor: Theme.palette.baseColor1
+
+   QtObject {
+        id: d
+
+        readonly property int inputTextPadding: 16
+        readonly property int radius: 8
+        readonly property int signingPhrasePadding: 8
+        readonly property int signingPhraseWordPadding: 8
+        readonly property int signingPhraseWordsSpacing: 8
+        readonly property int signingPhraseWordsHeight: 30
+
+    }
+
+    implicitWidth: 480
+    implicitHeight: 44
+
+    TextField {
```

Comment:
Please make the TextField the toplevel component, and just put the signing phrase repeater inside it

Reply:
You won't need to introduce all those toplevel properties